STN-157 : Date Stacked Horizontal UI Pattern

heatherdesigns opened the pull request:
# [STN-157](https://sparkbox.atlassian.net/browse/STN-157) READY FOR REVIEW

## Summary
This PR creates a `Date Stacked Horizontal` card designed to be used specifically for events.

This PR has been browser tested ✅, accessibility tested ✅, and reviewed by a designer ✅.

### 📝FYI 
The label style for the Date Stacked Horizontal UI Pattern is under design consideration by H&S. There may be an update to the color of the label.

## Need Review By (Date)
4/22/2020

## Urgency
medium

## Steps to Test
1. In the CLI, run `npm run test` and confirm our Sass tests pass.
1. Because this PR brings in a new pattern, you need to clear the cache by running `lando drush @economics.local cr`. This allows Drupal to recognize the new pattern.

### Add a new view
1. Go to the H&S/Sparkbox shared Google Drive Development directory and download an updated version of the `Custom Econ Events` view.
1. Go to Configuration > Development > Configuration Synchronization > Import
1. Click on the `Single item` tab, under `Configuration type` select `View`, paste settings from `custom-econ-events.txt`, and click `Import`.

### Create a basic page with the new view set to use the Date Stacked Horizontal pattern
1. Create a `Basic Page`
    * Content > Add Content > Basic Page
1. Add a `View` set to `Custom Econ Events` with a `Display` of `Date Horizontal - Single`
1. Save and check out your new page!

### Confirm that you can change the label styles
1. Edit the `Custom Econ Events` view `Date Horizontal - Single` display.
1. Under `Fields`, click on `Content: Speaker (Speaker)`
1. Under `Style Settings`, check `Customize field and label wrapper HTML`
1. Check `Create a CSS class` and add `hb-highlighted-label`
1. Apply and Save


## PR Checklist
- [PR Checklist](https://gist.github.com/sherakama/0ba17601381e3adbe0cad566ad4d80a5)
- [Sparkbox PR Checklist](../docs/SparkboxPRChecklist.md)

Commit:
feat: create date stacked horizontal card UI pattern

JIRA STN-157
https://sparkbox.atlassian.net/browse/STN-157

Commit:
feat: apply date stacked horizontal card styles

Commit:
feat: adjust pattern to not render drupal output when field data is empty

Commit:
feat: implement feedback from design pairing

Commit:
fix: IE alignment bug

- replace the use of postion: initial with position: static
- IE doesn't support position: initial and was causing the content to stack vertically rather than display horizontally

kaseybon reviewed (changes requested):
Everything looks good except the linter is failing.

Commit:
fix: linting error for empty line rule

bryanbraun commented:
Quick question on this, @heatherdesigns... what's the fallback behavior for when there's no image?

I was reminded of this when looking at one of the in-progress traditional designs below. These styles don't apply to the colorful theme, but I think the fallback behavior we see here is what they're hoping to achieve: 

![image](https://user-images.githubusercontent.com/1256329/80259121-d8258800-8652-11ea-8af8-52e795200d8d.png)

Commit:
feat: add calendar icon and styles for the date and time label and field

Commit:
feat: add location and speaker icons and their styles

Commit:
feat: update class name to be more reflective of use case

kaseybon commented on docroot/themes/humsci/humsci_basic/src/scss/components/_card.scss:
```diff
@@ -86,6 +86,67 @@
+      }
+
+      // only apply calendar icon when the label exists
+      .views-label-field-hs-event-date + .field-content {
```
We should add new classes for these icons fields (like `.hb-card__date-icon`) instead of targeting the `.views-label-` classes.

heatherdesigns replied:
I chatted with Bryan on this to make sure I was on the right path forward. Targeting the Drupal `.vews-label-` classes is fine but they needed to be scoped inside of a class specifically for the stying of the icons.

I created a `hb-card__icon` base class that has modifiers for `date`, `location`, and `speaker`. By pulling out the icon classes out from being nested under `hb-card__subcontent-detail` we make them available for placement in other twig templates which use the `hb-card` classes. (For future use cases.)

Commit:
feat: scope icons to an hb-card__icon class

- this creates flexibility should we decide to implement icons in other templates that use the hb-card classes

Commit:
feat: add compiled CSS after feedback changes
